docs/src/network_analysis/odes.md

```diff
@@ -0,0 +1,359 @@
+\frac{dm_2(t)}{dt} =& \frac{\alpha K^{n}}{K^{n} + \left( {P_1}\left( t \right) \right)^{n}} - \delta {m_2}\left( t \right) + \gamma \\
+\frac{dm_3(t)}{dt} =& \frac{\alpha K^{n}}{K^{n} + \left( {P_2}\left( t \right) \right)^{n}} - \delta {m_3}\left( t \right) + \gamma \\
+\frac{dP_1(t)}{dt} =& \beta {m_1}\left( t \right) - \mu {P_1}\left( t \right) \\
+\frac{dP_2(t)}{dt} =& \beta {m_2}\left( t \right) - \mu {P_2}\left( t \right) \\
+\frac{dP_3(t)}{dt} =& \beta {m_3}\left( t \right) - \mu {P_3}\left( t \right)
+\end{aligned}
+```
+
+## [Matrix-vector reaction rate equation representation](@id network_analysis_matrix_vector_representation)
+In general, reaction rate equation (RRE) ODE models for chemical reaction networks can
+be represented as a first-order system of ODEs in a compact matrix-vector notation. Suppose
+we have a reaction network with ``K`` reactions and ``M`` species, labelled by the state vector
+```math
+\mathbf{x}(t) = \begin{pmatrix} x_1(t) \\ \vdots \\ x_M(t)) \end{pmatrix}.
+```
+For the repressilator, ``\mathbf{x}(t)`` is just
+```@example s1
+x = species(repressilator)
+```
+The RRE ODEs satisfied by $\mathbf{x}(t)$ are then
+```math
+\frac{d\mathbf{x}}{dt} = N \mathbf{v}(\mathbf{x}(t),t),
+```
+where ``N`` is a constant ``M`` by ``K`` matrix with ``N_{m k}`` the net
+stoichiometric coefficient of species ``m`` in reaction ``k``.
+``\mathbf{v}(\mathbf{x}(t),t)`` is the rate law vector, with
+``v_k(\mathbf{x}(t),t)`` the rate law for the ``k``th reaction. For example,
+for the first reaction of the repressilator above, the rate law is
+```math
+v_1(\mathbf{x}(t),t) = \frac{\alpha K^{n}}{K^{n} + \left( P_3(t) \right)^{n}}.
+```
+We can calculate each of these in Catalyst via
+```@example s1
+N = netstoichmat(repressilator)
+```
+and by using the [`oderatelaw`](@ref) function
+```@example s1
+rxs = reactions(repressilator)
+ν = oderatelaw.(rxs)
+```
+Note, as [`oderatelaw`](@ref) takes just one reaction as input we use
+broadcasting to apply it to each element of `rxs`.
+
+Let's check that this really gives the same ODEs as Catalyst. Here is what Catalyst
+generates by converting to an `ODESystem`
+```@example s1
+osys = convert(ODESystem, repressilator)
+
+# for display purposes we just pull out the right side of the equations
+odes = [eq.rhs for eq in equations(osys)]
+```
+whereas our matrix-vector representation gives
+```@example s1
+odes2 = N * ν
+```
+Let's check these are equal symbolically
+```@example s1
+isequal(odes, odes2)
+```
+
+## [Reaction complex representation](@id network_analysis_reaction_complexes)
+We now introduce a further decomposition of the RRE ODEs, which has been used to
+facilitate analysis of a variety of reaction network properties. Consider a simple
+reaction system like
+```@example s1
+rn = @reaction_network begin
+ k, 2*A + 3*B --> A + 2*C + D
+ b, C + D --> 2*A + 3*B
```

Comment:
Think, generally, we have used `k, 2A + 3B --> A + 2C + D` (no `*`) across the docs. But feel free to use your preferred notation.